docs/literate/make.jl

```diff
@@ -0,0 +1,122 @@
+    # Searching for `[`Example`](@trixi-ref)` in content and replace it with `[`Example`](trixi_link/reference_trixi/#Trixi.Example)`.
+    content = replace(content, r"\[`(?<ref>\w+)`\]\(@trixi-ref\)" => SubstitutionString("[`\\g<ref>`]($(trixi_link)reference-trixi/#Trixi.\\g<ref>)"))
+end
+
+binder_logo   = "https://mybinder.org/badge_logo.svg"
+nbviewer_logo = "https://raw.githubusercontent.com/jupyter/design/master/logos/Badges/nbviewer_badge.svg"
+download_logo = "https://camo.githubusercontent.com/aea75103f6d9f690a19cb0e17c06f984ab0f472d9e6fe4eadaa0cc438ba88ada/68747470733a2f2f696d672e736869656c64732e696f2f62616467652f646f776e6c6f61642d6e6f7465626f6f6b2d627269676874677265656e"
+
+binder_url   = "https://mybinder.org/v2/gh/trixi-framework/TrixiTutorials/gh-pages?filepath=dev/notebooks"
+nbviewer_url = "https://nbviewer.jupyter.org/github/trixi-framework/TrixiTutorials/blob/gh-pages/dev/notebooks"
+download_url = "https://raw.githubusercontent.com/trixi-framework/TrixiTutorials/gh-pages/dev/notebooks"
+
+# Generate markdown for index.jl
+Literate.markdown(joinpath(repo_src, "index.jl"), pages_dir; name="introduction_literate", preprocess=preprocess_links,)
+# Navigation system for makedocs
+pages = Any["Introduction" => "tutorials/introduction_literate.md",]
+list = ["introduction_literate.md"]
+
+# Create markdown and notebook files for tutorials.
+for (i, (title, filename)) in enumerate(files)
+    # Several files of one topic are created seperately and pushed to `pages` together.
+    if filename isa Vector
+        vector = []
+        for j in 1:length(filename)
+            create_files("$i.$j: $title: $(filename[j][1])", filename[j][2][2]; folder=filename[j][2][1])
+
+            path = "$(filename[j][2][1])/$(splitext(filename[j][2][2])[1]).md"
+            push!(vector, "$i.$j $(filename[j][1])" => "tutorials/$path")
+            push!(list, path)
+        end
+        # Add to navigation menu
+        push!(pages, ("$i $title" => vector))
+    else # Single files
+        create_files("$i: $title", filename)
+        # Add to navigation menu
+        path = "$(splitext(filename)[1]).md"
+        push!(pages, ("$i $title" => "tutorials/$path"))
+        push!(list, path)
+    end
+end
+
+# Simple version of checking links
+for file in list
+    content = read(joinpath(pages_dir, file), String)
+    if occursin(r"\(https://[^\(\)]+\)", content)
+        matches = collect(eachmatch(r"\(https://[^\(\)]+\)", content))
+        for i in 1:length(matches)
+            link = string(chop(matches[i].match, head=1, tail=1))
+            try 
+                HTTP.get(link, retry=false, connect_timeout=15)
+            catch
+                if get(ENV, "CI", nothing) == "true"
+                    error("URL doesn't exist: ", link)
+                else
+                    @warn "URL doesn't exist: " link
+                end
+            end
+        end
+    end
+end
+
+return pages
```

Comment:
Does it make sense to wrap this in a function and use a `return` value for that function? It looks weird to `return` something from a file

Reply:
I agree with you, it really looks strange. Just wrapping all the code in a function, or you have a better idea?

Reply:
No, something like that sounds good to me